include/dlaf/permutations/general/impl.h

```diff
@@ -126,9 +129,51 @@ void applyPermutations(GlobalElementIndex out_begin, GlobalElementSize sz, SizeT
   }
 }
 
+template <class T, Device D, Coord C>
+void applyPermutationsFiltered(
+    GlobalElementIndex out_begin, GlobalElementSize sz, SizeType in_offset,
+    const matrix::Distribution& subm_dist, const SizeType* perm_arr,
+    const std::vector<pika::shared_future<matrix::Tile<const T, D>>>& in_tiles_fut,
+    const std::vector<matrix::Tile<T, D>>& out_tiles, std::function<bool(SizeType)> filter) {
+  constexpr auto OC = orthogonal(C);
+  std::vector<SizeType> splits =
+      dlaf::util::interleaveSplits(sz.get<OC>(), subm_dist.blockSize().get<OC>(),
+                                   subm_dist.distanceToAdjacentTile<OC>(in_offset),
+                                   subm_dist.distanceToAdjacentTile<OC>(out_begin.get<OC>()));
+
+  const SizeType nperms = subm_dist.size().get<C>();
+
+  // Parallelized over the number of permutations
+  pika::for_loop(pika::execution::par, to_sizet(0), to_sizet(nperms), [&](SizeType i_perm) {
```

Comment:
This is for a follow-up, i.e. no change requested here, but I want to comment to spread awareness: calls to pika's parallel algorithms should be replaced with the asynchronous versions. In this particular case we could simply use `bulk`.